.github/workflows/ci.yml

```diff
@@ -16,30 +16,52 @@ jobs:
         otp-version: ['22', '23', '24']
         # erlef/setup-beam action does not support macos yet
         os: [ubuntu-latest, windows-latest]
+        rebar3-version: ['3.17']
+        include:
+          - otp-version: '21'
+            os: ubuntu-latest
+            rebar3-version: '3.15.2'
+          - otp-version: '21'
+            os: windows-latest
+            rebar3-version: '3.15.2'
+          - otp-version: '20'
+            os: ubuntu-latest
+            rebar3-version: '3.15.2'
+
     runs-on: ${{ matrix.os }}
     env:
-      # Set to 1 for verbose rebar3 logging
-      DEBUG: 0
-      # Set to 1 for even more verbose rebar3 logging
-      DIAGNOSTIC: 0
+      # Define to something other than empty string to enable verbose rebar3 logging
+      DEBUG: ''
     steps:
       - name: Check out repository code
         uses: actions/checkout@v2
         with:
           persist-credentials: false
           submodules: recursive
       - name: Setup Erlang
+        id: setup-beam
         uses: ./.github/actions/setup-beam
         with:
           otp-version: ${{ matrix.otp-version }}
-          rebar3-version: '3.17'
+          rebar3-version: ${{ matrix.rebar3-version }}
       - name: Setup MSVC toolchain
         if: ${{ matrix.os == 'windows-latest' }}
         uses: ./.github/actions/msvc-dev-cmd
+      - name: Check coding style with erlfmt
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: rebar3 fmt --check
+        continue-on-error: true
       - name: Compile
         run: rebar3 compile
       - name: EUnit tests
         run: rebar3 eunit
+      - name: Restore dialyzer PLT from cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/rebar3/rebar3_*_plt
+          key: ${{ runner.os }}-${{ steps.setup-beam.outputs.otp-version }}
```

Comment:
So, all the documented examples for using the cache action include `runner.os` in the cache key, but that's actually just e.g. `Linux`. See actions/cache#543 for a request to have the specific OS version made available so the cache gets busted automatically when `ubuntu-latest` moves forward. In the meantime, a solution like [os-version-action](https://github.com/sersoft-gmbh/os-version-action) could be employed if this caching turns out to be a big issue.

The cache action saves 1-2 minutes of build time on this repo.

Reply:
Oh good find. That did take a bit to build locally